File: packages/typescript/src/roles/backend-developer.ts
import { Agent } from "../agent.js";
import { AgentRole, TaskType } from "../types.js";
import type { AgentConfig } from "../types.js";

const HANDLED_TASK_TYPES: TaskType[] = [
  TaskType.BackendFeature,
  TaskType.BugFix,
  TaskType.DBMigration,
  TaskType.CodeReview,
];

export class BackendDeveloper extends Agent {
  constructor(config: Partial<AgentConfig> = {}) {
    super({
      role: AgentRole.BackendDeveloper,
      ...config,
    });
  }

  canHandle(taskType: TaskType): boolean {
    return HANDLED_TASK_TYPES.includes(taskType);
  }
}

File: packages/python/src/tcs_bpi_ai_team/roles/backend_developer.py
"""Backend Developer agent."""

from __future__ import annotations

from tcs_bpi_ai_team.agent import Agent
from tcs_bpi_ai_team.types import AgentConfig, AgentRole, TaskType

_HANDLED_TASK_TYPES = [
    TaskType.BACKEND_FEATURE,
    TaskType.BUG_FIX,
    TaskType.DB_MIGRATION,
    TaskType.CODE_REVIEW,
]


class BackendDeveloper(Agent):
    """Senior backend developer — APIs, services, data access, tests."""

    def __init__(self, config: AgentConfig | None = None) -> None:
        super().__init__(config or AgentConfig(role=AgentRole.BACKEND_DEVELOPER))

    def can_handle(self, task_type: TaskType) -> bool:
        return task_type in _HANDLED_TASK_TYPES
